src/arithmetic.jl

```diff
@@ -280,10 +256,11 @@ function *{T<:NumberNotSeriesN}(a::HomogeneousPolynomial{T}, b::HomogeneousPolyn
 end
 
 
-# Homogeneous coefficient for the multiplication
+# Internal multiplication functions
 for T in (:Taylor1, :TaylorN)
-    @eval function mul!(c::$T, a::$T, b::$T, k::Int)
+    @eval @inline function mul!(c::$T, a::$T, b::$T, k::Int)
 
+        # c[k+1] = zero( a[k+1] )
         @inbounds for i = 0:k
             if $T == Taylor1
```

Comment:
We can certainly define a `mul` like method that returns `c[k+1] += a[i+1] * b[k-i+1]`. But my guess is that it will get inlined and produce the same code. Surely this will avoid the ugly code separating the `Taylor1` and `TaylorN` cases, but this happens at parse time, so I don't think we should care too much on this.